cluster/terraform_kubernetes/prometheus.tf

```diff
@@ -0,0 +1,164 @@
+resource "kubernetes_cluster_role" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name = "prometheus"
+  }
+
+  rule {
+    api_groups = [""]
+    resources  = ["nodes", "nodes/proxy", "services", "endpoints", "pods", ]
+    verbs      = ["get", "list", "watch", ]
+  }
+
+  rule {
+    api_groups = ["extensions", ]
+    resources  = ["ingresses", ]
+    verbs      = ["get", "list", "watch", ]
+  }
+
+  rule {
+    non_resource_urls = ["/metrics", ]
+    verbs             = ["get", ]
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "prometheus" {
+  count = var.enable_monitoring ? 1 : 0
+
+  metadata {
+    name = "prometheus"
+  }
+
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = "prometheus"
```

Comment:
Can we refer to the name in the terraform resource to enforce the dependency? ie `kubernetes_cluster_role.prometheus.metadata.name`

Reply:
done